pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -0,0 +1,147 @@
+package runtime
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/scheduler/framework"
+	frameworktesting "github.com/karmada-io/karmada/pkg/scheduler/framework/testing"
+)
+
+func Test_frameworkImpl_RunFilterPlugins(t *testing.T) {
+	ctx := context.Background()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockFilterPlugin := frameworktesting.NewMockFilterPlugin(mockCtrl)
+	tests := []struct {
+		name      string
+		mockFunc  func()
+		isSuccess bool
+	}{
+		{
+			name: "Test filter func ok",
+			mockFunc: func() {
+				mockFilterPlugin.EXPECT().Filter(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(framework.NewResult(framework.Success))
+			},
+			isSuccess: true,
+		},
+		{
+			name: "Test filter func error",
+			mockFunc: func() {
+				mockFilterPlugin.EXPECT().Filter(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(framework.NewResult(framework.Error, "foo error"))
+			},
+			isSuccess: false,
```

Comment:
It only test two cases: with one error or error plugin. Other cases are not covered: no plugin, mix plugins.

How about these:
```go
func Test_frameworkImpl_RunFilterPlugins(t *testing.T) {
	ctx := context.Background()
	mockCtrl := gomock.NewController(t)
	defer mockCtrl.Finish()

	alwaysError := frameworktesting.NewMockFilterPlugin(mockCtrl)
	alwaysError.EXPECT().Filter(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(framework.NewResult(framework.Error))

	alwaysSuccess := frameworktesting.NewMockFilterPlugin(mockCtrl)
	alwaysSuccess.EXPECT().Filter(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(framework.NewResult(framework.Success))

	tests := []struct {
		name      string
		framework framework.Framework
		isSuccess bool
	}{
		{
			name: "no plugin",
			framework: &frameworkImpl{
				filterPlugins: []framework.FilterPlugin{},
			},
			isSuccess: true,
		},
		{
			name: "error plugin",
			framework: &frameworkImpl{
				filterPlugins: []framework.FilterPlugin{alwaysError},
			},
			isSuccess: false,
		},
		{
			name: "success plugin",
			framework: &frameworkImpl{
				filterPlugins: []framework.FilterPlugin{alwaysSuccess},
			},
			isSuccess: true,
		},
		{
			name: "error and success plugins",
			framework: &frameworkImpl{
				filterPlugins: []framework.FilterPlugin{alwaysError, alwaysSuccess},
			},
			isSuccess: false,
		},
		{
			name: "success and error plugins",
			framework: &frameworkImpl{
				filterPlugins: []framework.FilterPlugin{alwaysSuccess, alwaysError},
			},
			isSuccess: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := tt.framework.RunFilterPlugins(ctx, nil, nil, nil)
			if result.IsSuccess() != tt.isSuccess {
				t.Errorf("want %v, but get:%v", tt.isSuccess, result.IsSuccess())
			}
		})
	}
}
```

Reply:
done for filter plugin, but not for score plugin, it does not increase code coverage and get few benefits.